fuzz/read-metadata.cpp

```diff
@@ -0,0 +1,65 @@
+// ***************************************************************** -*- C++ -*-
+// exifprint.cpp
+// Sample program to print the Exif metadata of an image
+
+#include <exiv2/exiv2.hpp>
+
+#include <iostream>
+#include <iomanip>
+#include <cassert>
+
+// https://github.com/Exiv2/exiv2/issues/468
+#if defined(EXV_UNICODE_PATH) && defined(__MINGW__)
+#undef  EXV_UNICODE_PATH
+#endif
+
+#ifdef  EXV_UNICODE_PATH
+#define _tchar      wchar_t
+#define _tstrcmp    wcscmp
+#define _t(s)       L##s
+#define _tcout      wcout
+#define _tmain      wmain
+#else
+#define _tchar      char
+#define _tstrcmp    strcmp
+#define _t(s)       s
+#define _tcout      cout
+#define _tmain      main
+#endif
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t * Data, size_t Size)
+try {
+    Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open(Data, Size);
+    assert(image.get() != 0);
+    image->readMetadata();
+
+    Exiv2::ExifData &exifData = image->exifData();
+    if (exifData.empty()) {
+        std::string error("No Exif data found in file");
+        throw Exiv2::Error(Exiv2::kerErrorMessage, error);
```

Comment:
Make this an early `return 0` instead, if the fuzzer produced an image without metadata, then that's hardly an error.